nengo_loihi/hardware/allocators.py

```diff
@@ -107,22 +117,57 @@ def one_to_one_allocator(model):
         for synapse in block.synapses:
             core.add_synapse(synapse)
 
-        for axon in block.axons:
-            core.add_axon(axon)
-
         stdp_pre_cfgs, stdp_pre_cfg_idxs = core_stdp_pre_cfgs(core)
         [core.add_stdp_pre_cfg(stdp_pre_cfg) for stdp_pre_cfg in stdp_pre_cfgs]
         core.stdp_pre_cfg_idxs = stdp_pre_cfg_idxs
 
         core.stdp_pre_profile_idx = None  # hardware.builder will set
         core.stdp_profile_idx = None  # hardware.builder will set
 
-    for input in model.inputs:
+    def input_to_chip(self, input, chip):
         # TODO: how to allocate inputs?
         core = chip.new_core()
         core.add_input(input)
-        for axon in input.axons:
-            core.add_axon(axon)
 
-    validate_board(board)
-    return board
+    def __call__(self, model):
+        board = Board()
+        chip = board.new_chip()
+
+        for block in model.blocks:
+            self.block_to_chip(block, chip)
+
+        for input in model.inputs:
+            self.input_to_chip(input, chip)
+
+        return board
+
+
+class RoundRobin(OneToOne):
```

Comment:
Does this help re-associate? https://en.wikipedia.org/wiki/Round-robin_scheduling

Reply:
Or this is actually more relevant because it's basically the same thing we're doing here: https://www.soundonsound.com/techniques/multiple-disk-drives

Reply:
Good to know the term is used in many places. The name still doesn't do anything to denote that this method is multi-chip, whereas `OneToOne` is not. But these will probably get renamed at some point anyway, because I don't see us having a single-chip one-to-one method long-term, since it would have no advantage over a greedy multi-chip method.

Reply:
Yeah I agree that multi-chip should become the norm / default and so it probably makes sense when that happens to rename `OneToOne`. It would just be there for sanity tests / debugging / profiling / etc.